components/stateTracker.js

```diff
@@ -90,6 +90,7 @@ StateTracker.prototype.bind = function() {
     var self = this;
     $(window).on('popstate', function() {
         if (!self.disabled) {
+            _.extend(history.state || {}, {_stateFromHistory: true});
```

Comment:
Actually, state depends on user behavior. Application does not depends on user behavior, application depends on its state.